crates/miden-lib/src/transaction/script_builder.rs

```diff
@@ -0,0 +1,146 @@
+    pub fn build_auth_script(&self) -> Result<TransactionScript, TransactionScriptBuilderError> {
+        self.build_script_with_sections(&[])
+    }
+
+    /// Builds a transaction script with the specified sections.
+    ///
+    /// The `sections` parameter is a vector of strings, where each string represents a distinct
+    /// part of the script body. The script includes, authentication, and expiration sections are
+    /// automatically added to the script.
+    fn build_script_with_sections(
+        &self,
+        sections: &[String],
+    ) -> Result<TransactionScript, TransactionScriptBuilderError> {
+        let script = format!(
+            "begin {} {} {} end",
+            self.script_expiration(),
+            sections.join(" "),
+            self.script_authentication()
+        );
+
+        let assembler = TransactionKernel::assembler().with_debug_mode(self.in_debug_mode);
+        let tx_script = TransactionScript::compile(script, vec![], assembler)
+            .map_err(TransactionScriptBuilderError::InvalidTransactionScript)?;
+
+        Ok(tx_script)
+    }
+
+    /// Returns a string with the authentication procedure call for the script.
+    fn script_authentication(&self) -> String {
+        let mut auth_script = String::new();
+        self.account_interface.auth().iter().for_each(|auth_scheme| match auth_scheme {
+            &AuthScheme::RpoFalcon512 { pub_key: _ } => {
+                auth_script
+                    .push_str("call.::miden::contracts::auth::basic::auth_tx_rpo_falcon512\n");
+            },
+        });
+
+        auth_script
+    }
+
+    /// Returns a string with the expiration delta update procedure call for the script.
+    fn script_expiration(&self) -> String {
+        if let Some(expiration_delta) = self.expiration_delta {
+            format!("push.{expiration_delta} exec.::miden::tx::update_expiration_block_delta\n")
+        } else {
+            String::new()
+        }
+    }
+}
+
+// TRANSACTION SCRIPT BUILDER ERROR
+// ============================================================================================
+
+/// Errors related to building a transaction script.
+#[derive(Debug, Error)]
+pub enum TransactionScriptBuilderError {
+    #[error("note asset is not issued by this faucet: {0}")]
+    IssuanceFaucetMismatch(AccountIdPrefix),
+    #[error("note created by the faucet doesn't contain exactly one asset")]
+    FaucetNoteWithoutAsset,
+    #[error("invalid transaction script")]
+    InvalidTransactionScript(#[source] TransactionScriptError),
+    #[error("invalid sender account: {0}")]
+    InvalidSenderAccount(AccountId),
+    #[error("{0} interface does not support the generation of the standard send_note script")]
+    UnsupportedInterface(AccountComponentInterface),
+    #[error("Account does not contain interfaces which support the send_note script generation")]
+    UnsupportedAccount,
```

Comment:
```suggestion
    #[error("account does not contain the basic fungible faucet or basic wallet interfaces which are needed to support the send_note script generation")]
    UnsupportedAccountInterface,
```

For clarity, we could mention the concrete interfaces that are supported.